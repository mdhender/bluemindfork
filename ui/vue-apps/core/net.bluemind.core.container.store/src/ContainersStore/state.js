export const state = {
    containers: {},
    containerKeys: []
};
