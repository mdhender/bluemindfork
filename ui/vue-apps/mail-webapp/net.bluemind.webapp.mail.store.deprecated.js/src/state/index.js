import { STATUS } from "../constants";

export let currentFolderKey;
export let foldersData = {};
export let login;
export let maxMessageSize;
export let messageFilter;
export let messagesWithUnblockedRemoteImages = [];
export let selectedMessageKeys = [];
export let showBlockedImagesAlert = false;
export const status = STATUS.IDLE;
export let userSettings = {};
