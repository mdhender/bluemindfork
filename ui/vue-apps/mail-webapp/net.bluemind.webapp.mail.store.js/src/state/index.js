export * from "./search";
export let currentFolderKey;
export let messageFilter;
export let foldersData = {};
export let login;
export let maxMessageSize;
