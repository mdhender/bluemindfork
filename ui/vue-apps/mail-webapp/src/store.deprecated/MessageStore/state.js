export default function () {
    return {
        id: undefined,
        key: undefined,
        parts: { attachments: [], inlines: [] },
        saveDate: null,
        status: null,
        attachmentStatuses: {},
        attachmentProgresses: {},
        to: [],
        cc: [],
        bcc: [],
        subject: "",
        content: "",
        isReplyExpanded: false,
        type: undefined,
        previousMessage: null,
        headers: []
    };
}
