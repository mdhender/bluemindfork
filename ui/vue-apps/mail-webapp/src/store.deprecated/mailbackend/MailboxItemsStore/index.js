import * as actions from "./actions";
import * as getters from "./getters";

export default {
    namespaced: true,
    actions,
    getters
};
