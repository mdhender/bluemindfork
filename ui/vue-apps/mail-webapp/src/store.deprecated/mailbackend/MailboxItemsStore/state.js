export const state = {
    items: {},
    partContents: {},
    itemKeys: [],
    itemsParts: {}
};
