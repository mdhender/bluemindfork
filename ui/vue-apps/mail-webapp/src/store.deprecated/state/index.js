import { STATUS } from "../constants";

export let userUid;
export let maxMessageSize;
export let messagesWithUnblockedRemoteImages = [];
export let showBlockedImagesAlert = false;
export const status = STATUS.IDLE;
