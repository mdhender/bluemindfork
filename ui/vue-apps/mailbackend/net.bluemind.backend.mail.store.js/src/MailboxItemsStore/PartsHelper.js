/** Utilities for body message parts handling. */
export default {
    insertInlineImages,
    isAttachment
};

/**
 * Replace the CID references found in partsWithReferences by the corresponding images found in imageParts.
 *
 * @param partsWithReferences parts with CID references inside an image markup
 * @param imageParts parts with base64 images and a CID identifier
 */
function insertInlineImages(partsWithReferences = [], imageParts = []) {
    const inlineReferenceRegex = /<img[^>]+?src\s*=\s*['"]cid:([^'"]*)['"][^>]*?>{1}?/gim;
    const inlined = [];
    partsWithReferences.forEach(partWithReferences => {
        let inlineReferences;
        while ((inlineReferences = inlineReferenceRegex.exec(partWithReferences.content)) !== null) {
            const cid = inlineReferences[1];
            const replaceRegex = new RegExp("(<img[^>]+?src\\s*=\\s*['\"])cid:" + cid + "(['\"][^>]*?>{1}?)", "gmi");
            const imagePart = imageParts.find(
                part => part.contentId && part.contentId.toUpperCase() === "<" + inlineReferences[1].toUpperCase() + ">"
            );
            if (imagePart) {
                partWithReferences.content = partWithReferences.content.replace(
                    replaceRegex,
                    "$1" + URL.createObjectURL(imagePart.content) + "$2"
                );
                inlined.push(imagePart.contentId);
            }
        }
    });
    return inlined;
}

/**
 * @return true if we consider the given part should be attached,
 *         i.e.: not shown in the message body, false otherwise
 */
function isAttachment(part) {
    return part.dispositionType && part.dispositionType === "ATTACHMENT";
}
