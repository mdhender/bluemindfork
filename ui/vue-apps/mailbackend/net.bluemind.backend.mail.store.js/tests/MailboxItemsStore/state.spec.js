import { state } from "../../src/MailboxItemsStore/state";

describe("[MailItemsStore][state]", () => {
    test("initial state", () => {
        expect(state).toMatchSnapshot();
    });
});
