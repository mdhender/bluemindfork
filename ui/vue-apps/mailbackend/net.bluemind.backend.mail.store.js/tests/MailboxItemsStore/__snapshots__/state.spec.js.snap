// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[MailItemsStore][state] initial state 1`] = `
Object {
  "itemKeys": Array [],
  "items": Object {},
  "itemsParts": Object {},
  "partContents": Object {},
}
`;
